### PR TITLE
Perf/read rpc message to end

### PR DESCRIPTION
**Pull request by LukaszRozmej**
## Changes

- Reads JSON RPC message to end before trying to deserialize it.
- Helps with performance on very long requests (which are fairly rare and less realistic scenarios)

#### What types of changes does your code introduce?

- [ ] Bugfix (a non-breaking change that fixes an issue)
- [ ] New feature (a non-breaking change that adds functionality)
- [ ] Breaking change (a change that causes existing functionality not to work as expected)
- [x] Optimization
- [ ] Refactoring
- [ ] Documentation update
- [ ] Build-related changes
- [ ] Other: _Description_

#### Requires testing

- [x] Yes
- [ ] No

#### If yes, did you write tests?

- [ ] Yes
- [x] No

#### Notes on testing

Would be worth comparing performance on very many batched requests (as many as you can under 30MB request limit) as it might be better or worse than original.

@marcindsobczak Would be worth comparing NewPayload performance with big but reasonable size blocks.

#### Requires documentation update

- [ ] Yes
- [x] No

#### Requires explanation in Release Notes

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/Nethermind/Nethermind.JsonRpc/JsonRpcProcessor.cs [hunk 1]
```diff
@@ -137,9 +137,7 @@ JsonRpcResult GetParsingError(string error, Exception? exception = null)
         try
         {
             // Asynchronously reads data from the PipeReader.
-            ReadResult readResult = _jsonRpcConfig.MaxRequestBodySize is not null
-                ? await reader.ReadAtLeastAsync((int)_jsonRpcConfig.MaxRequestBodySize.Value)
-                : await reader.ReadToEndAsync();
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.JsonRpc/JsonRpcProcessor.cs]: I'm torn about this one, @benaadams ?

Comment (benaadams) [reply src/Nethermind/Nethermind.JsonRpc/JsonRpcProcessor.cs]: Am wondering about adding a timeout also?

Comment (benaadams) [reply src/Nethermind/Nethermind.JsonRpc/JsonRpcProcessor.cs]: vs a very slow request; like if the full request takes more than 1sec to receive chop it?

Comment (LukaszRozmej) [reply src/Nethermind/Nethermind.JsonRpc/JsonRpcProcessor.cs]: We have some global timeouts on json rpc

**Review by benaadams — APPROVED** [review]

File: src/Nethermind/Nethermind.JsonRpc/JsonRpcProcessor.cs (hunk 1 above)
Comment (LukaszRozmej) [reply src/Nethermind/Nethermind.JsonRpc/JsonRpcProcessor.cs]: Will think about it a bit more.

**Pull request merged**